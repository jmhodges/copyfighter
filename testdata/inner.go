package main

import "net/http"

type someInt interface {
	Bang()
}
type bar struct {
	baz int
}

type other struct {
	quux int64
	srv  *http.Server
	si   someInt
}

func main() {
	type foo string
}

type Foo http.Client

func CallsFoo(f Foo) {

}

func (f Foo) OnOtherToo(o other) {

}

func (o other) OnStruct() {

}
func (o other) OnStruct2() {

}

func (o *other) OnPtr() {

}
func (o *other) OnPtr2() {

}
func (o *other) OnPtr3() {

}
