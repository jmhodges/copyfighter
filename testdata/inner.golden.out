# parameter 'f' at index 0 should be made into a pointer
func CallsFoo(f Foo)

# receiver, and parameter 'o' at index 0 should be made into pointers
func (Foo).OnOtherToo(o other)

# receiver should be made into a pointer
func (other).OnStruct()

# receiver should be made into a pointer
func (other).OnStruct2()

